File: core/src/androidTest/mock/com/mobilesolutionworks/android/controller/test/basic/SchedulerWorksFragment.java
package com.mobilesolutionworks.android.controller.test.basic;

import android.os.Bundle;

import com.mobilesolutionworks.android.app.WorksFragment;

/**
 * Created by yunarta on 12/3/17.
 */

public class SchedulerWorksFragment extends WorksFragment {

    public enum State {
        BEGIN,
        CREATE,
        RESUME,
        PAUSE,
        DESTROY
    }

    protected State mState = State.BEGIN;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        mState = State.CREATE;
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onPause() {
        mState = State.PAUSE;
        super.onPause();
    }

    @Override
    public void onResume() {
        mState = State.RESUME;
        super.onResume();
    }

    @Override
    public void onDestroy() {
        mState = State.DESTROY;
        super.onDestroy();
    }

    public State getState() {
        return mState;
    }
}

File: test-app/src/main/kotlin/com/mobilesolutionworks/android/controller/test/basic/SchedulerWorksFragment.kt
package com.mobilesolutionworks.android.controller.test.basic

import android.os.Bundle

import com.mobilesolutionworks.android.app.WorksFragment

/**
 * Created by yunarta on 12/3/17.
 */

open class SchedulerWorksFragment : WorksFragment() {

    enum class State {
        BEGIN,
        CREATE,
        RESUME,
        PAUSE,
        DESTROY
    }

    var state = State.BEGIN
        protected set

    override fun onCreate(savedInstanceState: Bundle?) {
        state = State.CREATE
        super.onCreate(savedInstanceState)
    }

    override fun onPause() {
        state = State.PAUSE
        super.onPause()
    }

    override fun onResume() {
        state = State.RESUME
        super.onResume()
    }

    override fun onDestroy() {
        state = State.DESTROY
        super.onDestroy()
    }
}
